diff --git a/sqlite-android/sqlite-android/src/main/jni/sqlite/android_database_SQLiteConnection.cpp b/sqlite-android/sqlite-android/src/main/jni/sqlite/android_database_SQLiteConnection.cpp
index d199c95..2872cca 100644
--- a/sqlite-android/sqlite-android/src/main/jni/sqlite/android_database_SQLiteConnection.cpp
+++ b/sqlite-android/sqlite-android/src/main/jni/sqlite/android_database_SQLiteConnection.cpp
@@ -950,11 +950,11 @@ static jboolean nativeHasCodec(JNIEnv* env, jobject clazz){
 #endif
 }
 
 static void nativeLoadExtension(JNIEnv* env, jobject clazz,
                                 jlong connectionPtr, jstring file, jstring proc) {
-    char* errorMessage;
+    /*char* errorMessage;
 
     SQLiteConnection* connection = reinterpret_cast<SQLiteConnection*>(connectionPtr);
     int result = sqlite3_enable_load_extension(connection->db, 1);
     if (result == SQLITE_OK) {
         const char* fileChars = env->GetStringUTFChars(file, NULL);
@@ -972,11 +972,11 @@ static void nativeLoadExtension(JNIEnv* env, jobject clazz,
         char* formattedError = sqlite3_mprintf("Could not register extension: %s", errorMessage);
         sqlite3_free(errorMessage);
 
         throw_sqlite3_exception_errcode(env, result, formattedError);
         sqlite3_free(formattedError);
-    }
+    }*/
 }
 
 static JNINativeMethod sMethods[] =
 {
     /* name, signature, funcPtr */
